database/database.go

```diff
@@ -0,0 +1,151 @@
+package database
```

Comment:
One thing you might not be doing that I thoroughly recommend is to run `go fmt` on your code before every push. Previously I've noticed a couple cases which `go fmt` would have cleaned up so at some point you haven't been doing this.